docs/upgrade.md

```diff
@@ -88,6 +88,22 @@ process, for example:
     dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
     ```
 
+# Upgrading to v1.81.0
+
+## Application service path & authentication deprecations
+
+Synapse now attempts the versioned appservice paths before falling back to the
+[legacy paths](https://spec.matrix.org/v1.6/application-service-api/#legacy-routes).
+Usage of the legacy routes should be considered deprecated.
+
+Additionally, Synapse has supported sending the application service access token
+via [the `Authorization` header](https://spec.matrix.org/v1.6/application-service-api/#authorization)
+since v1.70.0. For backwards compatibility it is *also* sent as the `access_token`
+query parameter. This is insecure and should be considered deprecated.
```

Comment:
Maybe including this in here doesn't make much sense since we theoretically wouldn't be compatible with [`v1.1` of the application service spec](https://spec.matrix.org/v1.1/application-service-api/#authorization) anymore.

(Weirdly the `r0.1.0` spec seems to provide no info on authorization.)

Anyway, I don't think Synapse declares it as supporting this version anywhere for the A-S API (unlike the C-S API which has a `/versions` endpoint).

Reply:
(spec version support is meant to be global to the project: servers declare v1.1 support across all APIs, for example)

Reply:
the query param was removed in v1.4, it looks like. Don't suppose the registration file contains the supported spec version or anything like that?

Reply:
It doesn't contain it, no.

I wonder if we should "deprecate" it and then in the future make you use a `enable_insecure_appservice_auth` config variable if you really want to keep doing it.

Reply:
> I would expect we don't have to support versions older than v1.4 forever, but I don't know who is allowed to make that decision

I've never heard talk of removing support for old versions, FWIW. 🤷 I definitely don't think we would drop support for <1.4 in a short-ish time frame.

Reply:
Another option is that we just punt out the query params bits and don't tie it up with the paths. 😄